src/containers/NFT/NFTHeader/NFTHeader.tsx

```diff
@@ -59,27 +105,33 @@ const NFTHeader = (props: Props) => {
   }
 
   const renderHeaderContent = () => {
-    const { issuer } = data
+    const { issuer } = data!
     return (
       <div className="section nft-header-container">
         <div className="nft-info-container">
           <div className="values">
             <div className="title">{t('issuer_address')}</div>
             <div className="value">
               <div className="nft-issuer">
-                <Account account={issuer} />
+                <Account account={issuer!} />
               </div>
             </div>
           </div>
         </div>
         <div className="nft-bottom-container">
           <div className="details">
             <div className="title">{t('details')}</div>
-            <Details data={data} />
+            <Details
+              data={{
+                ...data,
+                domain: accountData?.domain,
+                minted: mintedDate,
+              }}
+            />
           </div>
           <div className="settings">
             <div className="title">{t('settings')}</div>
-            <Settings flags={data.flags} />
+            <Settings flags={data?.flags ?? []} />
```

Comment:
isn't the `data!` syntax to let ts know that `data` is never null-ish? If that's true, then why would you need the `data?.flags` syntax instead of `data.flags`?